imports/ui/case-explorer/case-explorer.jsx

```diff
@@ -81,16 +86,15 @@ class CaseExplorer extends Component {
     (a, b) => a.length === b.length
   )
   makeCaseGrouping = memoizeOne(
-    (caseList, showOpen, onlyAssigned, allNotifs, unreadNotifs) => {
-      const openFilter = showOpen ? x => !isClosed(x) : x => isClosed(x)
-      const assignedFilter = onlyAssigned ? x => x.assignee === this.props.currentUser.bugzillaCreds.login : x => true
-
+    (caseList, roleFilterValues, sortBy, allNotifs, unreadNotifs) => {
+      const assignedFilter = roleFilterValues === 'All' ? x => true
+        : roleFilterValues === 'Assigned' ? x => x.assignee === this.props.currentUser.bugzillaCreds.login : x => true
       const caseUpdateTimeDict = this.makeCaseUpdateTimeDict(allNotifs)
       const caseUnreadDict = this.makeCaseUnreadDict(unreadNotifs)
 
       // Building a unit dictionary to group the cases together
-      const unitsDict = caseList.reduce((dict, caseItem) => {
-        if (openFilter(caseItem) && assignedFilter(caseItem)) { // Filtering only the cases that match the selection
+      const unitsDict = caseList.sort(sorters[sortBy]).filter(caseItem => !isClosed(caseItem)).reduce((dict, caseItem) => {
```

Comment:
The `sort()` should be done last. No point of sorting items that get filtered out anyway

Reply:
Look at line 119. It appears that cases are never sorted by the proper selected sort, but always by the latest updated logic. I'll add a separate comment about that, since I can see that the bug is easily reproduceable.